server/src/main/java/org/opensearch/index/IndexService.java

```diff
@@ -524,13 +525,16 @@ public synchronized IndexShard createShard(
                 remoteStore = new Store(shardId, this.indexSettings, remoteDirectory, lock, Store.OnClose.EMPTY);
             }
 
-            TranslogFactory translogFactory = this.indexSettings.isRemoteTranslogStoreEnabled() && routing.primary()
-                ? new RemoteBlobStoreInternalTranslogFactory(
-                    repositoriesServiceSupplier,
-                    threadPool,
-                    this.indexSettings.getRemoteStoreTranslogRepository()
-                )
-                : new InternalTranslogFactory();
+            BiFunction<IndexSettings, ShardRouting, TranslogFactory> translogFactorySupplier = (idxSettings, shardRouting) -> {
```

Comment:
Can we define this supplier only once? Nothing specific to the shard getting created, right?